### DISCUSSION TITLE
How smart is it?

### DISCUSSION THREAD

**Discussion by dabrahams** [SwiftyLab/setup-swift#231] [Q&A] [answered]
I notice that when I ask for swift 5.9 on GitHub's `macos-latest` runner, this action installs Swift.  Is that because the runner has an earlier version, or is there a missing optimization here?

Thanks!

P.S. Love the very usable interface to installing on any platform.  Thanks for making this available.

**Comment by soumyamahunt** [comment, marked as answer]
Glad to know you find this action useful. I just noticed this discussion so here is the rationale.

Swift 5.9 requires Xcode 15 as mentioned in [swift.org](https://www.swift.org/download/). The `macos-latest` runner is listed to be [using `macos-12` runner as of now](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-private-repositories), [which doesn't have Xcode 15 installed](https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#xcode). Hence the action downloads Swift 5.9.

This action won't download Swift if required Xcode version is installed and will just switch the default Xcode version to that installation. i.e. for `macos-13` Swift 5.9 won't be installed, [you can also see this in action as part of integration test](https://github.com/SwiftyLab/setup-swift/actions/runs/8107841107/job/22160040552).

**Comment by dabrahams** [comment]
Yeah, I eventually found out macos-latest was 12, not 13 (how suprising!)  Sorry for not closing this issue.

That said, I don't think

> Swift 5.9 requires Xcode 15 

is true.  You can install a Swift 5.9 release on a machine with no Xcode installed at all.  In fact, I'm pretty sure Xcode 15 doesn't run on macOS 12, so when your action downloads Swift 5.9 on `macos-latest` that Swift runs without Xcode 15 too.

**Reply by soumyamahunt** [reply]
Agree that "requires" is the wrong word to use here, what I meant was Xcode 15 comes with Swift 5.9 and in presence of Xcode 15 it won't be reinstalled by this action.
